### PR TITLE
Use first letter of name and colored circle as user icon in timeline

### PR DESCRIPTION
**Pull request by mvidalgarcia**
Current look:
image

### CODE REVIEW

File: indico/htdocs/js/indico/modules/abstracts.js [hunk 1]
```diff
@@ -15,7 +15,8 @@
  * along with Indico; if not, see <http://www.gnu.org/licenses/>.
  */
 
-/* global setupListGenerator:false, setupSearchBox:false, applySearchFilters:false, Palette:false */
+/* global setupListGenerator:false, setupSearchBox:false, applySearchFilters:false, Palette:false,
+          avatarPalette: false */
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/abstracts.js]: Does this actually work? The last time I tried wrapping the line it did not recognize the ones from the new lines so I had to repeat the `/* global ` part

File: indico/htdocs/js/indico/modules/abstracts.js [hunk 2]
```diff
@@ -80,6 +81,27 @@
...
+        function hashCode(str) {
+            var hash = 0;
+            var i;
+            if (!str.length) {
+                return hash;
+            }
+            for (i = 0; i < str.length; i++) {
+                var char = str.charCodeAt(i);
+                hash = ((hash << 5) - hash) + char;
+                hash = hash & hash;
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/abstracts.js]: What's that line supposed to do?

File: indico/htdocs/js/palette.js [hunk 3]
```diff
@@ -27,4 +27,9 @@
 
     // colors for specific purposes
     Palette.highlight = Palette.blue;
+
+    // colors for the bubble avatar in abstract page
+    global.avatarPalette = ['#E06055', '#FF8A65', '#E91E63', '#F06292', '#3F51B5', '#BA68C8', '#7986CB', '#5E97f6',
```
Comment (ThiefMaster) [inline indico/htdocs/js/palette.js]: Can we use all uppercase or (even better imo) all lowercase for the hex chars and not mix both please? :)

Comment (ThiefMaster) [inline indico/htdocs/js/palette.js]: I'd go for `Palette.userColors = [...]` instead of using another global.

File: indico/htdocs/js/palette.js [hunk 4]
```diff
@@ -27,4 +27,9 @@
 
     // colors for specific purposes
     Palette.highlight = Palette.blue;
+
+    // colors for the bubble avatar in abstract page
```
Comment (ThiefMaster) [inline indico/htdocs/js/palette.js]: `// colors for user-specific avatar bubbles`

We might reuse this in other places so I'd rather not mention abstracts

File: indico/modules/events/abstracts/templates/abstract/public.html [hunk 5]
```diff
@@ -425,7 +427,9 @@ <h3 class="abstract-title f-self-grow">
 
             {% if abstract.state.name not in ('submitted', 'withdrawn') %}
                 <div class="i-timeline-item">
-                    <div class="i-timeline-item-label action"></div>
+                    <div class="i-timeline-item-label action user-avatar" data-username="{{ abstract.judge.name }}">
+                        {{- abstract.judge.name[0] -}}
```
Comment (ThiefMaster) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Don't we want `display_full_name` here to get the user's preferred name display style?

**Comment by ThiefMaster** [comment]
Considering that we might want to support proper avatars in the future, maybe it'd be better if we added an `avatar_css` property to the `User` model that for now returns a `background-color: #...` based on the username. Then you'd just use `style="{{ whatever_user.avatar_css }}" in the template and not need JavaScript for it at all.

File: indico/htdocs/js/indico/modules/abstracts.js (hunk 1 above)
Comment (mvidalgarcia) [inline indico/htdocs/js/indico/modules/abstracts.js]: It was working for me but it won't be problem anymore since we will be using `Palette.userColors`

**Comment by OmeGak** [comment]
Beautiful! :lipstick:

File: indico/htdocs/js/palette.js (hunk 4 above)
Comment (OmeGak) [inline indico/htdocs/js/palette.js]: I'd rather not add the comment. Isn't `avatarPalette` clear enough?

File: indico/htdocs/sass/modules/abstracts/_abstract_page.scss [hunk 6]
```diff
@@ -215,15 +215,9 @@
...
-        background: white !important;
-        color: $light-black !important;
-        justify-content: flex-end !important;
-
-        &:before {
-            font-size: 2.5em !important;
-        }
+        color: rgba(#fff, 0.8);
+        font-size: 1.65em;
+        border-radius: 1em;
```
Comment (OmeGak) [inline indico/htdocs/sass/modules/abstracts/_abstract_page.scss]: I think it's better specifying `50%` here.

File: indico/modules/events/abstracts/templates/abstract/public.html (hunk 5 above)
Comment (OmeGak) [inline indico/modules/events/abstracts/templates/abstract/public.html]: Not necessarily. This is Indico choosing which big letter to show.

**Comment by mvidalgarcia** [comment]
Logic moved to backend

File: indico/modules/users/util.py [hunk 7]
```diff
@@ -347,3 +352,7 @@ def merge_users(source, target, force=False):
     db.session.flush()
 
     logger.info("Successfully merged %s into %s", source, target)
+
+
+def get_color_hash_by_str(s):
```
Comment (ThiefMaster) [inline indico/modules/users/util.py]: `get_color_for_username`

**Comment by ThiefMaster** [comment]
Preview of the current palette: https://jsfiddle.net/3yeL2jh7/

**Comment by pferreir** [comment]
LGTM

**Pull request merged**